### ISSUE TITLE
Default CI Java Version to Java 11, run tests on 8, 14 and 17

### ISSUE THREAD

**Issue by dblock** [opensearch-project/index-management#200]
Coming from https://github.com/opensearch-project/opensearch-plugins/issues/64, default CI Java version to 11 (LTS), and run tests on 8, 14 and 17.

- [x] Change the default Java version for CI in GitHub actions from 14/15 to 11 (LTS).
- [ ] Add Java 8, 14 (used before), and 17 (next LTS) to the CI matrix.
- [x] Document Java versions in DEVELOPER_GUIDE.md.

**Comment by dblock** [comment]
@downsrob I believe you only have JDK8 left to do for 1.3.0 here.

**Comment by downsrob** [comment]
Just built with JDK8 and got a compilation failure
```
e: /home/runner/work/index-management/index-management/src/main/kotlin/org/opensearch/indexmanagement/indexstatemanagement/step/forcemerge/WaitForForceMergeStep.kt: (62, 56): Type mismatch: inferred type is Long but BigDecimal! was expected
```
@dblock Do we need build support for JDK 8 in the 1.3 release? As in we will need to get this fixed on the 1.3 branch by the end of the day?

**Comment by downsrob** [comment]
There were a couple of small lines to change to get JDK 8 building. Not sure if this is something I should backport after the code freeze date, so I just have it in a draft PR here https://github.com/opensearch-project/index-management/pull/297
